### PR TITLE
 when clicking on skip credits, skip to the next episode #4 

### PR DESCRIPTION
**Pull request by dredstone1**
until now when i clicked on skip credit it skip to the end of the video and then i needed to click on next up, this fix that, so when i click on skip credits button it will just skip to the next episode.
this meant to mimic the behavior of the web-version
the same as #4

### CODE REVIEW

**Comment by rlauuzo** [comment]
There might be content after the credits. You'll need to check if the credits end within 3 seconds of the episode's total runtime (and hope the user doesn't change the intro playback duration from 2 seconds to something else).

**Comment by jumoog** [comment]
> There might be content after the credits. You'll need to check if the credits end within 3 seconds of the episode's total runtime (and hope the user doesn't change the intro playback duration from 2 seconds to something else).

yes i can remember.. i did this mistake as well once

**Comment by dredstone1** [comment]
> There might be content after the credits. You'll need to check if the credits end within 3 seconds of the episode's total runtime (and hope the user doesn't change the intro playback duration from 2 seconds to something else).

I will change it, so it will only move to the next episode if it's end in the last 3s of the episode and if there even is a next episode,

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt [hunk 1]
```diff
@@ -46,7 +46,11 @@ class SegmentSkipFragment() : Fragment() {
 
 	private fun buttonClicked() {
 		lastSegment?.let { segment ->
-			playbackControllerContainer.playbackController?.seek((segment.endTime.millis).toLong())
+			if (segment.type == SegmentType.CREDITS && playbackControllerContainer.playbackController?.hasNextItem() == true) {
```
Comment (AbandonedCart) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt]:
```
playbackControllerContainer.playbackController?.let { player ->
    if (segment.type == SegmentType.CREDITS && player.hasNextItem() == true) {
        player.next()
    } else {
        player.seek(segment.endTime.millis)
    }
}
```
Saves a bunch of redundant null checks

Comment (AbandonedCart) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt]: `git fetch upstream; git rebase upstream/master; git push -f` and you won't need to use `toLong` for millis anymore

Comment (jumoog) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt]: > ```
> playbackControllerContainer.playbackController?.let { player ->
>     if (segment.type == SegmentType.CREDITS && player.hasNextItem() == true) {
>         player.next()
>     } else {
>         player.seek(segment.endTime.millis)
>     }
> }
> ```
> Saves a bunch of redundant null checks

To be fair he just adopted the code style

Comment (AbandonedCart) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt]: Yeah. I also didn't change any of that.

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt [hunk 2]
```diff
@@ -46,7 +46,13 @@ class SegmentSkipFragment() : Fragment() {
 
 	private fun buttonClicked() {
 		lastSegment?.let { segment ->
-			playbackControllerContainer.playbackController?.seek((segment.endTime.millis).toLong())
+			playbackControllerContainer.playbackController?.let { player ->
+				if (((segment.endTime.millis+180000) >= player.getDuration()) && player.hasNextItem() == true) {
```
Comment (AbandonedCart) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt]: (segment.endTime + 5).millis

Comment (AbandonedCart) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt]: So
```
if ((segment.endTime + 5).millis >= player.getDuration() && player.hasNextItem() == true)
```

**Comment by jumoog** [comment]
its a draft **OR** ready for review

**Comment by dredstone1** [comment]
it was ready and then i saw that i did a mistake and instead of 3s is 3m

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt (hunk 2 above)
Comment (dredstone1) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt]: why 180 and not 5000?

**Comment by AbandonedCart** [comment]
Now you have 5 seconds

**Comment by dredstone1** [comment]
> Now you have 5 seconds

it 5000 but it 3s,

**Comment by AbandonedCart** [comment]
1000 milliseconds per second

**Comment by dredstone1** [comment]
> 1000 milliseconds per second

i know, but when i wrote 3000 it did only +1s

**Comment by AbandonedCart** [comment]
Problem isn’t the metric system, though.

**Comment by dredstone1** [comment]
when i did log it wrote that the length of the episode is 2s more the it really was
i will do some more test

**Comment by dredstone1** [comment]
its because the skip is skipping 2s before the determined time is

**Comment by AbandonedCart** [comment]
Seems like accounting for a user setting with a finite padding is dangerous.

**Comment by dredstone1** [comment]
i will fix it

**Comment by dredstone1** [comment]
i see that in the injected intro skipper button its also have this problem but it uses 3s so when the credit is  ending 2s before the end it will not auto skip to the next episode, so in that case if no one complain about it not really being 3s i will also use 3000

**Comment by AbandonedCart** [comment]
By injected, you mean the web server?

**Comment by dredstone1** [comment]
> By injected, you mean the web server?

yes, the web cliient

**Comment by AbandonedCart** [comment]
Mirroring that code is your best option. It should work the same way across platforms.

**Comment by dredstone1** [comment]
exactly, this is the point of this pr

**Pull request merged**